pkg/api/store/password/store.go

```diff
@@ -109,86 +124,120 @@ func (p *PasswordStore) Watch(apiContext *types.APIContext, schema *types.Schema
 	return p.Stores[schema.ID].Watch(apiContext, schema, opt)
 }
 
-type fieldInfo struct {
-	paths []string
-	value string
-}
-
-func (p *PasswordStore) replacePasswords(data map[string]interface{}, id string, existing map[string]interface{}) error {
-	var fieldData []fieldInfo
-	var path []string
-	buildFieldData(convert.ToMapInterface(p.Fields[id]), data, &fieldData, path)
-
-	return p.handlePasswordFields(fieldData, data, id, existing)
-}
-
-func (p *PasswordStore) assignBack(data map[string]interface{}, id string, delete bool) error {
-	var fieldData []fieldInfo
-	var path []string
-
-	buildFieldData(convert.ToMapInterface(p.Fields[id]), data, &fieldData, path)
+func (p *PasswordStore) replacePasswords(sepData, data, existing map[string]interface{}) error {
+	/*
+		sepData: path to all password fields built recursively from schema
+		data: incoming data from api
+		existing: data from db by id
 
-	for _, info := range fieldData {
-		split := strings.SplitN(info.value, ":", 2)
-		if len(split) != 2 {
-			continue
-		}
-		if delete {
-			p.deleteSecret(split[1], split[0])
-		} else {
-			value, err := p.getSecret([]string{split[0], split[1]})
-			if err != nil {
-				return fmt.Errorf("error getting secret for field %s", info.value)
+		Looping through sepData,
+			if separator == reached end, check to update/create or replace password fields
+			else recursive call for map/array
+	*/
+	for key1, val1 := range sepData {
+		if convert.ToString(val1) == separator {
+			if val2, ok := data[key1]; ok {
+				if err := p.putSecretData(data, existing, key1, convert.ToString(val2), false); err != nil {
+					logrus.Errorf("errr %v", err)
+				}
+			} else if _, ok := existing[key1]; ok {
+				if err := p.putSecretData(data, existing, key1, "", true); err != nil {
+					logrus.Errorf("errr %v", err)
+				}
+			} else if data != nil {
+				logrus.Warnf("[%v] not present in incoming data, secret not stored", key1)
```

Comment:
Question: why do we need this warning? because the if on line 139 covers the case when incoming data has the password/secret type field. And else if on line 143 covers when it's not in the incoming data but is in the existing data.

Reply:
When data is neither in incoming data nor existing, 
this can happen when 
- during create, secretKey value is not passed 
- during update, uniqueKey got changed and new secretKey wasn't passed 

I'm putting this warning for the second case, though it might be extraneous for second